Add greedyCollapse option

simonasdev opened the pull request:
**Describe the change**

Add an option to make deflating less greedy

**Further comments**

Some very long horizontal or vertical polygons get deflated too soon

oliverroick commented:
Hi @simonasdev, thanks for the PR. This seems like a useful addition. 

Before I can merge and cut a new release, would you be able to:

- fix the lint issues,
- add tests, and
- add documentation?

I can help with those but I'm not sure when I'll get to it

simonasdev commented:
I've fixed lint issues and added documentation, but not yet sure how tests should be implemented. I guess I would need a fixture specific for this new option

oliverroick commented:
Thanks for making these changes.

For testing, the best way to to this is to:

- Add a fixture [here](https://github.com/oliverroick/Leaflet.Deflate/blob/master/tests/fixtures.js#L119)
- Add a couple of tests, [similar to these](https://github.com/oliverroick/Leaflet.Deflate/blob/master/tests/L.Deflate.test.js#L328-L363)

I can have a look at those over the weekend, if you need support

oliverroick commented:
I wasn't able to push the tests to your branch. I'm merging the PR and add the tests separately
